aws/sdk/integration-tests/s3/tests/express.rs

```diff
@@ -0,0 +1,249 @@
+                    <Expiration>2024-01-29T18:53:01Z</Expiration>
+                </Credentials>
+            </CreateSessionResult>
+            "#,
+        ))
+        .unwrap()
+}
+
+#[tokio::test]
+async fn presigning() {
+    let http_client = StaticReplayClient::new(vec![ReplayEvent::new(
+        create_session_request(),
+        create_session_response(),
+    )]);
+
+    let config = aws_sdk_s3::Config::builder()
+        .http_client(http_client)
+        .region(Region::new("us-west-2"))
+        .with_test_defaults()
+        .build();
+    let client = Client::from_conf(config);
+
+    let presigning_config = PresigningConfig::builder()
+        .start_time(SystemTime::UNIX_EPOCH + Duration::from_secs(1234567891))
+        .expires_in(Duration::from_secs(30))
+        .build()
+        .unwrap();
+
+    let presigned = client
+        .get_object()
+        .bucket("s3express-test-bucket--usw2-az1--x-s3")
+        .key("ferris.png")
+        .presigned(presigning_config)
+        .await
+        .unwrap();
+
+    let uri = presigned.uri().parse::<Uri>().unwrap();
+
+    let pq = uri.path_and_query().unwrap();
+    let path = pq.path();
+    let query = pq.query().unwrap();
+    let mut query_params: Vec<&str> = query.split('&').collect();
+    query_params.sort();
+
+    pretty_assertions::assert_eq!(
+        "s3express-test-bucket--usw2-az1--x-s3.s3express-usw2-az1.us-west-2.amazonaws.com",
+        uri.authority().unwrap()
+    );
+    assert_eq!("GET", presigned.method());
+    assert_eq!("/ferris.png", path);
+    pretty_assertions::assert_eq!(
+        &[
+            "X-Amz-Algorithm=AWS4-HMAC-SHA256",
+            "X-Amz-Credential=ASIARTESTID%2F20090213%2Fus-west-2%2Fs3express%2Faws4_request",
+            "X-Amz-Date=20090213T233131Z",
+            "X-Amz-Expires=30",
+            "X-Amz-S3session-Token=TESTSESSIONTOKEN",
+            "X-Amz-Signature=c09c93c7878184492cb960d59e148af932dff6b19609e63e3484599903d97e44",
+            "X-Amz-SignedHeaders=host",
+            "x-id=GetObject"
+        ][..],
+        &query_params
+    );
+    assert_eq!(presigned.headers().count(), 0);
+}
+
+#[tokio::test]
+async fn disable_s3_express_session_auth_at_service_client_level() {
+    let (http_client, request) = capture_request(None);
+    let conf = Config::builder()
+        .http_client(http_client)
+        .region(Region::new("us-west-2"))
+        .with_test_defaults()
+        .disable_s3_express_session_auth(true)
+        .build();
+    let client = Client::from_conf(conf);
+
+    let _ = client
+        .list_objects_v2()
+        .bucket("s3express-test-bucket--usw2-az1--x-s3")
+        .send()
+        .await;
+
+    let req = request.expect_request();
+    assert!(
+        !req.headers()
+            .get("authorization")
+            .unwrap()
+            .contains("x-amz-create-session-mode"),
+        "x-amz-create-session-mode should not appear in headers when S3 Express session auth is disabled"
```

Comment:
we probably want an S3 express section added to the canary. We'll need to update the CDK for the canary to create the bucket. That's where I'd probably smoke test all of the APIs

Reply:
Yeah, I was planning to add an S3 express test for canary, but at the same time I'd like to mainly exercise the happy path use cases. A failure in canary will block our release, so I want to reduce the amount of things we do in canary as much as possible. For testing a negative case like disabling S3 Express, I'd still want to keep in an integration test.